### PR TITLE
added verbatim support, work around for jsx or any other body injection

### PR DESCRIPTION
**Pull request by Grimace1975**
var $ = program('');
$.body.append('function () {' + $.verbatim('Anything I Want') + '}');
var newProgram = $.toString();

### CODE REVIEW

**Comment by SBoudrias** [comment]
Can you add unit tests?

**Comment by Grimace1975** [comment]
Yes I can add some unit tests.
My approach maybe overkill also.

Ideally trying to support embeded JSX encode, decode. Tried different options, enabling jsx on latest esprima.  esprima with jsx parsed most of my JSX unless there was code in the element. But escodegen could not reconstruct back to javascript.

This is a simple injection approach. Using the verbatim support in escodegen.

Looking at perhaps changing ast-query to acorn or acorn-jsx, and escodegen or escodegen-jsx, based on options.

**Comment by Grimace1975** [comment]
@SBoudrias have some questions.. Added acorn support, but included this verbatim methods. 

Should I make a separate pull request.

Also do not have a single configuration point, since tree is not passed around everywhere.
To select between esprima and acorn, would have to pass as option in each entry point (main, valuefactory, body)

**Comment by SBoudrias** [comment]
@Grimace1975 couldn't we just move to acorn and drop esprima?

**Comment by Grimace1975** [comment]
@SBoudrias yes. I was not sure how much we wanted to leave the linage. I have also found error reporting is different. I have made it optional, by adding references back to the original instance, so one option should switch everything.

**Comment by SBoudrias** [comment]
@Grimace1975 let's move everything to acorn, we'll just bump a major version.

**Comment by Grimace1975** [comment]
@SBoudrias ok, simplifies things. tests where going to be an issue too. but not now.

**Comment by Grimace1975** [comment]
Adding some tests for jsx, and updating the Readme.md

**Comment by Grimace1975** [comment]
@SBoudrias everything is done. Looks like an issue with parsing JSX the first time, but there is a work around. Commented out the test. added a separate test.

**Comment by Grimace1975** [comment]
and for some reason, still can parse the spread operator below:

        let property = this.props.property;
        return {...property};

**Comment by Grimace1975** [comment]
found a bug. adding test

File: README.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ Program
 ### `var tree = program( sourceCode, escodegenOptions, esprimaOptions )`
 - **sourceCode** (String) - The source code to edit.
 - **escodegenOptions** (Object) _optional_ - [escodegen](https://github.com/Constellation/escodegen) option object
-- **esprimaOptions** (Object) _optional_ - [esprima](http://esprima.org/doc) option object
+- **acronOptions** (Object) _optional_ - [acron](https://github.com/ternjs/acorn) option object
```
Comment (SBoudrias) [inline README.md]: typo `s/acron/acorn/`

**Comment by SBoudrias** [comment]
One typo, otherwise it looks good to go!

**Comment by Grimace1975** [comment]
Cool. Ya. Did you fix the typo. I might not have done such a great job on the markdown.

Also interestingly enough I am building the same library for c# ast

Sky Morey
Sent from my iPhone

On Jun 9, 2016, at 11:11 PM, Simon Boudrias <notifications@github.com<mailto:notifications@github.com>> wrote:


One typo, otherwise it looks good to go!

-

**Comment by SBoudrias** [comment]
@Grimace1975 if you want, I can make you a contributor on the repository. I'm not actively working on it, so if you'd like to improve it further it'd be glad to open the door.

**Comment by Grimace1975** [comment]
Sure.

Also. I made a separate project. AST-query.edge for the c# version. Using edge to get an ast tree. Then using in node. Then back to edge to get the code for it. It's kinda cool.

Sky Morey
Sent from my iPhone

On Jun 9, 2016, at 11:18 PM, Simon Boudrias <notifications@github.com<mailto:notifications@github.com>> wrote:


@Grimace1975<https://github.com/Grimace1975> if you want, I can make you a contributor on the repository. I'm not actively working on it, so if you'd like to improve it further it'd be glad to open the door.

-

**Comment by SBoudrias** [comment]
Alright, merged!

**Pull request closed without merging**